### PR TITLE
rateQuality file

### PR DESCRIPTION
**Pull request by saulmezac**
First PR of rateQuality

### CODE REVIEW

File: pyautomagic/src/rateQuality.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+def rateQuality(OHA, THV, CHV, RCB, overallGoodCutoff:float=0.1, overallBadCutoff:float=0.2, timeGoodCutoff:float=0.1, timeBadCutoff:float=0.2,
+                channelGoodCutoff:float=0.15, channelBadCutoff:float=0.3, BadChannelGoodCutoff:float=0.15, BadChannelBadCutoff:float=0.3):
+    
+    # check that thresholds are between 0 and 1
+    if any(isinstance(x, int) for x in [fill_this_in]) or any(x <= 0 or x >= 1 for x in [fill_this_in]):
+           logger.log(f"Some threshold cutoffs were set as integer. Please pass in a float between 0 and 1. You passed in {stuff}."
+           raise ValueError("Say something")
+
+    qualityScores = [OHA, THV, CHV, RCB]
+    Qs = qualityScores
```
Comment (adam2392) [inline pyautomagic/src/rateQuality.py]: You are doubly naming things.

File: pyautomagic/src/rateQuality.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    
+    # check that thresholds are between 0 and 1
+    if any(isinstance(x, int) for x in [fill_this_in]) or any(x <= 0 or x >= 1 for x in [fill_this_in]):
+           logger.log(f"Some threshold cutoffs were set as integer. Please pass in a float between 0 and 1. You passed in {stuff}."
+           raise ValueError("Say something")
+
+    qualityScores = [OHA, THV, CHV, RCB]
+    Qs = qualityScores
+
+    # Categorize wrt OHA
```
Comment (adam2392) [inline pyautomagic/src/rateQuality.py]: Create a separate function:

def something(qs, threshold):
        <the if/else to return good,okay,bad>
       return string

**Comment by adam2392** [comment]
@clairezurn you can review this code sometime and see if you have any constructive criticism, or summarizing why it seems to work.

**Comment by clairezurn** [comment]
Seems like it would work just fine, and looks similar to the matlab one. I'm less concerned about docstring style right now and more would like to see better python style in the code. Use more descriptive variables (acronyms are bad) and all should be non-capitalized first letter. Also, rather than just putting up an error if inputs are wrong, why not set it to the default and let the user know? That's what I'm doing for calcQuality. This all in addition to what Adam said above.

**Comment by saulmezac** [comment]
> 1. docstring accordingly with the style of: https://www.sphinx-doc.org/en/1.5/ext/example_google.html
> 2. You have no return statement. I'm not sure how you're even checking to see if this worked. In addition, there are like... 20 if/else statements. This is completely unreadable and unable to be debugged.
> 3. Once you fix the above, add a unit test.

The changes were done Adam.

**Comment by clairezurn** [comment]
The rateQuality function looks great now. Test file needs more cases. Why are you only testing the basic input? Test to make sure that it does what you want it to when perfect input isn't there. See mine/others for examples.

**Comment by saulmezac** [comment]
> The rateQuality function looks great now. Test file needs more cases. Why are you only testing the basic input? Test to make sure that it does what you want it to when perfect input isn't there. See mine/others for examples.

Done.

**Comment by AamnaLawrence** [comment]
The code is very readable. Good use of comments!

**Comment by saulmezac** [comment]
> Sure this is fine for now, but just FYI. You should include in the comments details about what the function is doing.
> 
> https://github.com/NeuroDataDesign/automagic/blob/master/src/rateQuality.m
> 
> For example, it should be clear that the rating is the worst of a set of ratings combined. {good, good, good, bad} = {bad}
> 
> {good, good, good, good} = {good}.

Done, thanks Adam.

**Pull request merged**